#include "contestwindow.h"
#include "ui_contestwindow.h"
#include"mainwindow.h"
contestwindow::contestwindow(QWidget *parent)
    : QWidget(parent)
    , ui(new Ui::contestwindow)
{
    ui->setupUi(this);
    contestCurrentLine = new QLineEdit;

    setWindowTitle("ContestMenu");

    if(!main)
    {
        main = std::make_unique<MainWindow>();
    }
    \
    connect(ui->pushButtonHelp, &QPushButton::clicked, this, &contestwindow::pushHelp);
    connect(ui->pushButtonCurrent,&QPushButton::clicked,this,&contestwindow::pushCurrentButton);
    connect(ui->pushButtonAttestat,&QPushButton::clicked,this,&contestwindow::pushAttestatButton);
    connect(ui->pushButtonBackMain,&QPushButton::clicked,this,&contestwindow::BackMenu);

}

contestwindow::~contestwindow()
{
    delete ui;
}

void contestwindow::pushHelp()
{

    QString helpText =
        "üìò <b>–ö–æ–Ω–∫—É—Ä—Å—ã:</b><br>"
        "- Sportakiada<br>"
        "- certificate<br>"
        "- Hackathon<br><br>"
        "üìÑ <b>–ö–æ–Ω–∫—É—Ä—Å –ê—Ç—Ç–µ—Å—Ç–∞—Ç–æ–≤:</b><br>"
        "–î–ª—è –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏—è —ç—Ç–æ–≥–æ –º–µ–Ω—é –≤—ã –¥–æ–ª–∂–Ω—ã –∏–º–µ—Ç—å –≥–æ—Ç–æ–≤—ã–π —Ñ–∞–π–ª —Å–æ —Å—Ç—É–¥–µ–Ω—Ç–∞–º–∏ –≤ —Ñ–æ—Ä–º–∞—Ç–µ:<br>"
        "<i>–ò–º—è –ë–∞–ª–ª</i>";

    QMessageBox msgBox;
    msgBox.setWindowTitle("–ü–æ–º–æ—â—å");
    msgBox.setText(helpText);
    msgBox.setWindowIcon(QIcon("C:/Users/Gera/Documents/Student_Manager_ARDA/ico/iconInfo.png"));
    msgBox.exec();

}

void contestwindow::pushCurrentButton()
{
    DialogContest dialog(ModeCurrent, this);

    if(dialog.exec() == QDialog::Accepted)
    {
    }
}

void contestwindow::pushAttestatButton()
{
    DialogContest dialog(ModeAttestat, this);

    if(dialog.exec() == QDialog::Accepted);
}

void contestwindow::BackMenu()
{
    main->show();
    this->close();

}
